### ISSUE TITLE
Bug: compiling vsproj file

### ISSUE THREAD

**Issue by WouterVandenputte** [bdero/flutter_scene#30]
When attempting to run the flutter scene importer using the following command:

```
dart --enable-experiment=native-assets run flutter_scene_importer:import --input "DamagedHelmet.glb" --output "DamagedHelmet.model" 
``` 

All I get is a huge ton of error output that I do not seem to find anyone else having this problem. So I thought to report it here. All I can see is that is trying to compile a .vsproj file.  I attached both the output log of that command and the `flutter doctor` below.
Please let me know if I am doing something wrong on my part and/or if this is not the right place to post this.

Snippet output log (full log in file due to Github post length constraint):

> PS D:\Users\woute\Repositories\test3d> dart --enable-experiment=native-assets run flutter_scene_importer:import --input "DamagedHelmet.glb" --output "DamagedHelmet.model"
> Unhandled exception:
> Exception: CMake build step failed (exit code 1):
> STDERR:
> STDOUT: Microsoft (R) Build Engine version 15.9.21+g9802d43bc3 for .NET Framework
> 
> Copyright (C) Microsoft Corporation. All rights reserved.
> 
> 
> 
>   1>Checking Build System
> 
>   tinygltf.vcxproj -> C:\Users\woute\AppData\Local\Pub\Cache\hosted\pub.dev\flutter_scene_importer-0.1.2-3\build\_deps\tinygltf-build\Debug\tinygltf.lib
> 
>   flatc.vcxproj -> C:\Users\woute\AppData\Local\Pub\Cache\hosted\pub.dev\flutter_scene_importer-0.1.2-3\build\_deps\flatbuffers-build\Debug\flatc.exe
> 
>   Generating flatbuffer schema C:/Users/woute/AppData/Local/Pub/Cache/hosted/pub.dev/flutter_scene_importer-0.1.2-3/scene.fbs
> 
>   conversions.cc
> 
> c:\users\woute\appdata\local\pub\cache\hosted\pub.dev\flutter_scene_importer-0.1.2-3\types.h(39): error C3615: constexpr function 'impeller::scene::importer::Matrix::Matrix' cannot result in a constant expression [C:\Users\woute\AppData\Local\Pub\Cache\hosted\pub.dev\flutter_scene_importer-0.1.2-3\build\importer.vcxproj]
> 
>   c:\users\woute\appdata\local\pub\cache\hosted\pub.dev\flutter_scene_importer-0.1.2-3\types.h(39): note: failure was caused by 'constexpr' constructor not initializing member 'impeller::scene::importer::Matrix::<anonymous-tag>'
> 
>   c:\users\woute\appdata\local\pub\cache\hosted\pub.dev\flutter_scene_importer-0.1.2-3\types.h(39): note: failure was caused by 'constexpr' constructor not initializing member 'impeller::scene::importer::Matrix::<anonymous-tag>'
> 
>   c:\users\woute\appdata\local\pub\cache\hosted\pub.dev\flutter_scene_importer-0.1.2-3\types.h(39): note: failure was caused by 'constexpr' constructor not initializing member 'impeller::scene::importer::Matrix::<anonymous-tag>'
> 
> c:\users\woute\appdata\local\pub\cache\hosted\pub.dev\flutter_scene_importer-0.1.2-3\types.h(46): error C3615: constexpr function 'impeller::scene::importer::Matrix::Matrix' cannot result in a constant expression [C:\Users\woute\AppData\Local\Pub\Cache\hosted\pub.dev\flutter_scene_importer-0.1.2-3\build\importer.vcxproj]
> 
>   c:\users\woute\appdata\local\pub\cache\hosted\pub.dev\flutter_scene_importer-0.1.2-3\types.h(46): note: failure was caused by 'constexpr' constructor not initializing member 'impeller::scene::importer::Matrix::<anonymous-tag>'
> 
>   c:\users\woute\appdata\local\pub\cache\hosted\pub.dev\flutter_scene_importer-0.1.2-3\types.h(46): note: failure was caused by 'constexpr' constructor not initializing member 'impeller::scene::importer::Matrix::<anonymous-tag>'
> 
>   c:\users\woute\appdata\local\pub\cache\hosted\pub.dev\flutter_scene_importer-0.1.2-3\types.h(46): note: failure was caused by 'constexpr' constructor not initializing member 'impeller::scene::importer::Matrix::<anonymous-tag>'
> 
>   importer_gltf.cc
> 
> c:\users\woute\appdata\local\pub\cache\hosted\pub.dev\flutter_scene_importer-0.1.2-3\types.h(39): error C3615: constexpr function 'impeller::scene::importer::Matrix::Matrix' cannot result in a constant expression [C:\Users\woute\AppData\Local\Pub\Cache\hosted\pub.dev\flutter_scene_importer-0.1.2-3\build\importer.vcxproj]
> 
>   c:\users\woute\appdata\local\pub\cache\hosted\pub.dev\flutter_scene_importer-0.1.2-3\types.h(39): note: failure was caused by 'constexpr' constructor not initializing member 'impeller::scene::importer::Matrix::<anonymous-tag>'
> 
>   c:\users\woute\appdata\local\pub\cache\hosted\pub.dev\flutter_scene_importer-0.1.2-3\types.h(39): note: failure was caused by 'constexpr' constructor not initializing member 'impeller::scene::importer::Matrix::<anonymous-tag>'
> 
...
[output_log.txt](https://github.com/user-attachments/files/17265751/output_log.txt)

```
flutter doctor -v
````

> [√] Flutter (Channel master, 3.26.0-1.0.pre.213, on Microsoft Windows [Version 10.0.19045.4894], locale nl-BE)
>     • Flutter version 3.26.0-1.0.pre.213 on channel master at D:\flutter
>     • Upstream repository https://github.com/flutter/flutter.git
>     • Framework revision f310625bc6 (10 days ago), 2024-09-25 10:19:00 -0700
>     • Engine revision d6d5fdba6a
>     • Dart version 3.6.0 (build 3.6.0-278.0.dev)
>     • DevTools version 2.40.0-dev.1
> 
> [√] Windows Version (Installed version of Windows is version 10 or higher)
> 
> [!] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
>     • Android SDK at C:\Users\woute\AppData\Local\Android\sdk
>     X cmdline-tools component is missing
>       Run `path/to/sdkmanager --install "cmdline-tools;latest"`
>       See https://developer.android.com/studio/command-line for more details.
>     X Android license status unknown.
>       Run `flutter doctor --android-licenses` to accept the SDK licenses.
>       See https://flutter.dev/to/windows-android-setup for more details.
> 
> [√] Chrome - develop for the web
>     • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe
> 
> [!] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.8.4)
>     • Visual Studio at D:\Program Files\Microsoft Visual Studio\2022\Community
>     • Visual Studio Community 2022 version 17.8.34408.163
>     • Windows 10 SDK version 10.0.17763.0
>     X Visual Studio is missing necessary components. Please re-run the Visual Studio installer for the "Desktop development with C++" workload, and include these components:
>         MSVC v142 - VS 2019 C++ x64/x86 build tools
>          - If there are multiple build tool versions available, install the latest
>         C++ CMake tools for Windows
>         Windows 10 SDK
> 
> [√] Android Studio (version 2024.1)
>     • Android Studio at D:\Program Files\Android\Android Studio
>     • Flutter plugin can be installed from:
>        https://plugins.jetbrains.com/plugin/9212-flutter
>     • Dart plugin can be installed from:
>        https://plugins.jetbrains.com/plugin/6351-dart
>     • Java version OpenJDK Runtime Environment (build 17.0.11+0--11852314)
> 
> [√] VS Code (version 1.93.1)
>     • VS Code at C:\Users\woute\AppData\Local\Programs\Microsoft VS Code
>     • Flutter extension version 3.25.0
> 
> [√] Connected device (3 available)
>     • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19045.4894]
>     • Chrome (web)      • chrome  • web-javascript • Google Chrome 127.0.6533.89
>     • Edge (web)        • edge    • web-javascript • Microsoft Edge 128.0.2739.42
> 
> [√] Network resources
>     • All expected network resources are available.

**Comment by WouterVandenputte** [comment]
@kwill39 done

**Comment by bdero** [comment]
This is similar to https://github.com/bdero/flutter_scene/issues/14.

I just pushed a commit to remove the constexpr qualifiers in the importer. Could you try again against commit https://github.com/bdero/flutter_scene/commit/76192b065841e502b15c926334548a29479d59d1 and let me know if there are any more compilation errors?

In your pubspec:
```yaml
dependencies:
  flutter_scene:
    git:
      url: https://github.com/bdero/flutter_scene.git
      ref: 76192b065841e502b15c926334548a29479d59d1
```

**Comment by WouterVandenputte** [comment]
First of all, thanks for reaching out so quickly.

I added that specific git commit as a dependency, I had to do some overrides to make package resolving succeed. Hopefully this is correct.
![image](https://github.com/user-attachments/assets/4b9baef1-3d72-4ec7-96b8-0ae071625b48)

I then did a clean, get and run. **The errors now are different**. I'll include the entire log again in a txt file, but from what I can tell, there's only one error that is constantly repeating.

> C:\Users\woute\AppData\Local\Pub\Cache\git\flutter_scene-76192b065841e502b15c926334548a29479d59d1\importer\build\_deps\flatbuffers-src\src\idl_parser.cpp(1,1): error C1041: cannot open program database 'C:\Users\woute\AppData\Local\Pub\Cache\git\flutter_scene-76192b065841e502b15c926334548a29479d59d1\importer\build\_deps\flatbuffers-build\flatc.dir\Debug\vc143.pdb'; if multiple CL.EXE write to the same .PDB file, please use /FS [C:\Users\woute\AppData\Local\Pub\Cache\git\flutter_scene-76192b065841e502b15c926334548a29479d59d1\importer\build\_deps\flatbuffers-build\flatc.vcxproj]

[log2.txt](https://github.com/user-attachments/files/17269191/log2.txt)

**Comment by bdero** [comment]
I'm not sure what's going on there. If repeated builds fail, perhaps try deleting `C:\Users\woute\AppData\Local\Pub\Cache\git` to clear out the generated VS project and other intermediaries.

I fired up Windows to give it a try and ran into another pathing issue that is now fixed with https://github.com/bdero/flutter_scene/commit/4f5a2475cc24362402c5129657e661de42fe7e6f .

I'm able to get the example app running with Impeller enabled, but unfortunately nothing displays and the app freezes, so it looks like there's still a bug or two that I need to work out with Flutter GPU on Windows.

![image](https://github.com/user-attachments/assets/d41b41dc-3f9f-4582-b105-0cb3607396a8)
